## Using Loop

In [11]:
def factorial (n):
    a= 1
    for i in range (1, n + 1):
        a *=  i 
    print(a)

In [12]:
if __name__ == "__main__":
    
    n = int(input("Enter a number for getting factorial"))
    print (f"User want {n}'s factorial")
    
    factorial (n)

User want 5's factorial
120


## Using Recursion 

In [15]:
def factorial_no (n1):
    
    #Handling 0 and 1 recursion stoppage 
    if n1 == 1 or n1 == 0 :
        return 1
    else :
        return (n1 * factorial_no(n1-1))

n1 = int(input("Enter a number for getting factorial"))
print (f"User want {n1}'s factorial")
factorial_no (n1)

User want 5's factorial


120

### Main Recursion Logic

- 5 * factorial_no 4 
- 5 * 4 * factorial_no 3 
- 5 * 4 * 3 * factorial_no 2
- 5 * 4 * 3 * 2 factorial_no 1
- 5 * 4 * 3 * 2 * 1 = 120